Failed to parse NVD data - CvssV4Data error?

Our vulnerability scans have recently started returning the following error and failing our CICD jobs. Is this a known issue?

I am running version 12.0.2 and java 21

```
[2025-02-15 21:02:02.042] [ERROR] org.owasp.dependencycheck.Engine - Error updating the NVD Data
org.owasp.dependencycheck.data.update.exception.UpdateException: Error updating the NVD Data
	at org.owasp.dependencycheck.data.update.NvdApiDataSource.processApi(NvdApiDataSource.java:399)
	at org.owasp.dependencycheck.data.update.NvdApiDataSource.update(NvdApiDataSource.java:117)
	at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:903)
	at org.owasp.dependencycheck.Engine.initializeAndUpdateDatabase(Engine.java:708)
	at org.owasp.dependencycheck.Engine.analyzeDependencies(Engine.java:634)
	at nvd.task.check$scan_and_analyze$fn__793.invoke(check.clj:52)
	at nvd.task.check$scan_and_analyze.invokeStatic(check.clj:51)
	at nvd.task.check$scan_and_analyze.invoke(check.clj:46)
	at nvd.task.check$impl.invokeStatic(check.clj:89)
	at nvd.task.check$impl.invoke(check.clj:81)
	at nvd.task.check$_main.invokeStatic(check.clj:148)
	at nvd.task.check$_main.doInvoke(check.clj:100)
	at clojure.lang.RestFn.invoke(RestFn.java:424)
	at clojure.lang.Var.invoke(Var.java:390)
	at user$eval138.invokeStatic(form-init10682709095142075290.clj:1)
	at user$eval138.invoke(form-init10682709095142075290.clj:1)
	at clojure.lang.Compiler.eval(Compiler.java:7700)
	at clojure.lang.Compiler.eval(Compiler.java:7690)
	at clojure.lang.Compiler.load(Compiler.java:8165)
	at clojure.lang.Compiler.loadFile(Compiler.java:8103)
	at clojure.main$load_script.invokeStatic(main.clj:476)
	at clojure.main$init_opt.invokeStatic(main.clj:478)
	at clojure.main$init_opt.invoke(main.clj:478)
	at clojure.main$initialize.invokeStatic(main.clj:509)
	at clojure.main$null_opt.invokeStatic(main.clj:543)
	at clojure.main$null_opt.invoke(main.clj:540)
	at clojure.main$main.invokeStatic(main.clj:665)
	at clojure.main$main.doInvoke(main.clj:617)
	at clojure.lang.RestFn.applyTo(RestFn.java:140)
	at clojure.lang.Var.applyTo(Var.java:707)
	at clojure.main.main(main.java:40)
Caused by: io.github.jeremylong.openvulnerability.client.nvd.NvdApiException: Failed to parse NVD data
	at io.github.jeremylong.openvulnerability.client.nvd.NvdCveClient._next(NvdCveClient.java:389)
	at io.github.jeremylong.openvulnerability.client.nvd.NvdCveClient.next(NvdCveClient.java:357)
	at org.owasp.dependencycheck.data.update.NvdApiDataSource.processApi(NvdApiDataSource.java:355)
	... 30 more
Caused by: com.fasterxml.jackson.databind.exc.ValueInstantiationException: Cannot construct instance of `io.github.jeremylong.openvulnerability.client.nvd.CvssV4Data$ModifiedCiaType`, problem: SAFETY
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 1283671] (through reference chain: io.github.jeremylong.openvulnerability.client.nvd.CveApiJson20["vulnerabilities"]->java.util.ArrayList[405]->io.github.jeremylong.openvulnerability.client.nvd.DefCveItem["cve"]->io.github.jeremylong.openvulnerability.client.nvd.CveItem["metrics"]->io.github.jeremylong.openvulnerability.client.nvd.Metrics["cvssMetricV40"]->java.util.ArrayList[0]->io.github.jeremylong.openvulnerability.client.nvd.CvssV4["cvssData"]->io.github.jeremylong.openvulnerability.client.nvd.CvssV4Data["modifiedSubsequentSystemIntegrity"])
```

I'm getting this error in the docker environment, v7.0.2.

Caused by: java.lang.IllegalArgumentException: SAFETY
        at io.github.jeremylong.openvulnerability.client.nvd.CvssV4Data$ModifiedCiaType.fromValue(CvssV4Data.java:1224)
        at jdk.internal.reflect.GeneratedMethodAccessor20.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:569)
        at com.fasterxml.jackson.databind.introspect.AnnotatedMethod.callOnWith(AnnotatedMethod.java:118)
        at com.fasterxml.jackson.databind.deser.std.FactoryBasedEnumDeserializer.deserialize(FactoryBasedEnumDeserializer.java:194)
        ... 66 common frames omitted

Unable to complete NVD cache update due to error: Failed to parse NVD data
io.github.jeremylong.vulnz.cli.cache.CacheException: Unable to complete NVD cache update due to error: Failed to parse NVD data
        at io.github.jeremylong.vulnz.cli.commands.CveCommand.processRequest(CveCommand.java:286)
        at io.github.jeremylong.vulnz.cli.commands.CveCommand.timedCall(CveCommand.java:229)
        at io.github.jeremylong.vulnz.cli.commands.TimedCommand.call(TimedCommand.java:36)
        at io.github.jeremylong.vulnz.cli.commands.TimedCommand.call(TimedCommand.java:25)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2045)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2465)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2457)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2419)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2421)
        at picocli.CommandLine.execute(CommandLine.java:2174)
        at io.github.jeremylong.vulnz.cli.Application.run(Application.java:73)
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:765)
        at org.springframework.boot.SpringApplication.lambda$callRunners$2(SpringApplication.java:749)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:510)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:744)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1300)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1289)
        at io.github.jeremylong.vulnz.cli.Application.main(Application.java:61)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:569)
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65)

Completed in 17 seconds


2025-02-16 08:33:18,192 WARN exited: init_nvd_cache (exit status 1; not expected)

--- a/core/src/main/java/org/owasp/dependencycheck/data/nvdcve/CveDB.java
+++ b/core/src/main/java/org/owasp/dependencycheck/data/nvdcve/CveDB.java
@@ -929,20 +929,20 @@ public final class CveDB implements AutoCloseable {
                         if (value != null) {
                             modifiedVulnAvailabilityImpact = CvssV4Data.ModifiedCiaType.fromValue(value);
                         }
-                        CvssV4Data.ModifiedCiaType modifiedSubConfidentialityImpact = null;
+                        CvssV4Data.ModifiedSubCType modifiedSubConfidentialityImpact = null;
                         value = rsV.getString(56);
                         if (value != null) {
-                            modifiedSubConfidentialityImpact = CvssV4Data.ModifiedCiaType.fromValue(value);
+                            modifiedSubConfidentialityImpact = CvssV4Data.ModifiedSubCType.fromValue(value);
                         }
-                        CvssV4Data.ModifiedCiaType modifiedSubIntegrityImpact = null;
+                        CvssV4Data.ModifiedSubIaType modifiedSubIntegrityImpact = null;
                         value = rsV.getString(57);
                         if (value != null) {
-                            modifiedSubIntegrityImpact = CvssV4Data.ModifiedCiaType.fromValue(value);
+                            modifiedSubIntegrityImpact = CvssV4Data.ModifiedSubIaType.fromValue(value);
                         }
-                        CvssV4Data.ModifiedCiaType modifiedSubAvailabilityImpact = null;
+                        CvssV4Data.ModifiedSubIaType modifiedSubAvailabilityImpact = null;
                         value = rsV.getString(58);
                         if (value != null) {
-                            modifiedSubAvailabilityImpact = CvssV4Data.ModifiedCiaType.fromValue(value);
+                            modifiedSubAvailabilityImpact = CvssV4Data.ModifiedSubIaType.fromValue(value);
                         }
                         CvssV4Data.SafetyType safety = null;
                         value = rsV.getString(59);
@@ -2191,6 +2191,22 @@ public final class CveDB implements AutoCloseable {
     }
 
     private void setUpdateColumn(PreparedStatement ps, int i, CvssV4Data.ModifiedCiaType value) throws SQLException  {
+        if (value == null) {
+            ps.setNull(i, java.sql.Types.VARCHAR);
+        } else {
+            ps.setString(i, value.value());
+        }
+    }
+
+    private void setUpdateColumn(PreparedStatement ps, int i, CvssV4Data.ModifiedSubCType value) throws SQLException  {
+        if (value == null) {
+            ps.setNull(i, java.sql.Types.VARCHAR);
+        } else {
+            ps.setString(i, value.value());
+        }
+    }
+
+    private void setUpdateColumn(PreparedStatement ps, int i, CvssV4Data.ModifiedSubIaType value) throws SQLException  {
         if (value == null) {
             ps.setNull(i, java.sql.Types.VARCHAR);
         } else {

--- a/pom.xml
+++ b/pom.xml
@@ -932,7 +932,7 @@ Copyright (c) 2012 - Jeremy Long
             <dependency>
                 <groupId>io.github.jeremylong</groupId>
                 <artifactId>open-vulnerability-clients</artifactId>
-                <version>7.2.1</version>
+                <version>7.2.2</version>
             </dependency>
             <dependency>
                 <groupId>org.anarres.jdiagnostics</groupId>

